File: python/leetcode/implement-magic-dictionary.py
class MagicDictionary:

    def __init__(self):
        """
        Initialize your data structure here.
        """
        self.lookup = []
        

    def buildDict(self, dict):
        """
        Build a dictionary through a list of words
        :type dict: List[str]
        :rtype: void
        """
        for word in dict:
            self.lookup.append(word)      

    def search(self, word):
        """
        Returns if there is any word in the trie that equals to the given word after modifying exactly one character
        :type word: str
        :rtype: bool
        """
        for w in self.lookup:
            if w == word or len(w) != len(word):
                continue
            diff = 0
            for i in range(len(word)):
                if word[i] != w[i]:
                    diff += 1
                if diff > 1:
                    break
            if diff == 1:
                return True
        return False

# Your MagicDictionary object will be instantiated and called as such:
# obj = MagicDictionary()
# obj.buildDict(dict)
# param_2 = obj.search(word)

File: javascript/leetcode/implement-magic-dictionary.js
/**
 * Initialize your data structure here.
 */
var MagicDictionary = function() {
    this.lookup = []
};

/**
 * Build a dictionary through a list of words 
 * @param {string[]} dict
 * @return {void}
 */
MagicDictionary.prototype.buildDict = function(dict) {
    this.lookup = dict
};

/**
 * Returns if there is any word in the trie that equals to the given word after modifying exactly one character 
 * @param {string} word
 * @return {boolean}
 */
MagicDictionary.prototype.search = function(word) {
    this.lookup.forEach(function(w) {
        if (w !== word && w.length == word.length) {
            diff = 0
            for (var i = 0; i < word.length; i++) {
                c1 = word[i]
                c2 = word[i]
                if (c1 !== c2) {
                    diff++
                }
                if (diff > 1) {break}
            }
            if (diff == 1) {
                return true
            }
        }
    })
    return false
};

/** 
 * Your MagicDictionary object will be instantiated and called as such:
 * var obj = Object.create(MagicDictionary).createNew()
 * obj.buildDict(dict)
 * var param_2 = obj.search(word)
 */